support/util/networking.go

```diff
@@ -121,3 +121,29 @@ func AllowedCIDRBlocks(hcp *hyperv1.HostedControlPlane) []hyperv1.CIDRBlock {
 	}
 	return nil
 }
+
+func SetAdvertiseAddress(hcp *hyperv1.HostedControlPlane, defaults *DefaultAdvIps) string {
```

Comment:
This function returns an advertise address, so it seems a more appropriate name would be `GetAdvertiseAddress`

Reply:
In general, this function is distinguishing between ipv4 and ipv6 and then essentially doing what `AdvertiseAddressWithDefault` does, so maybe you should call that function from here.